libnmstate/nm/nmclient.py

```diff
@@ -44,7 +43,15 @@ def client(refresh=False):
     # refresh is a workaround to get the current state when GMainLoop is not
     # running
     if _nmclient is None or refresh:
-        _nmclient = NM.Client.new(None)
+        if NM:
+            _nmclient = NM.Client.new(None)
+        else:
+            logging.error(
+                'Missing introspection data for libnm'
+                'please make sure to install it prior to running nmstate.\n'
+                'Check the documentation for more information.'
+            )
+            exit(1)
```

Comment:
IMHO it is wrong to exit here. This would for example kill an ansible proceess instead of allowing it to properly report back an error message.

Reply:
I had no better idea and focused on a cli client. Any suggestion? You prefer an exception?

Reply:
I'll use an exception.